### ISSUE TITLE
[Error]Sorry, looks like your network settings are preventing access to this feature.

### ISSUE THREAD

**Issue by kagangtuya-star** [dice2o/BingGPT#25]
This error message suddenly appeared today.

"Sorry, looks like your network settings are preventing access to this feature."

I'm in an area that newbing doesn't yet provide, but I still get this error when I turn on a global agent or system agent to another area.

![Snipaste_2023-03-24_10-24-55](https://user-images.githubusercontent.com/73411104/227408772-9d686f67-dc8b-426a-9f6e-2336a5d078b3.png)

**Comment by Marshall1115** [comment]
I have just encountered this problem.  i think it seems to software problem

**Comment by kagangtuya-star** [comment]
> I have just encountered this problem. i think it seems to software problem
I seem to have found a solution where changing the DNS settings to 1.1.1.1
But it's not stable.

**Comment by Marshall1115** [comment]
> nice, I have also just found this way to solve this problem, just set clash VPN proxy mode to global mode.

**Comment by kagangtuya-star** [comment]
> > nice, I have also just found this way to solve this problem, just set clash VPN proxy mode to global mode.

Perhaps it is some kind of commercial competition? 1.1.1.1 The provider of this dns is CloudFlare, which has a good relationship with Microsoft. If bing's competitors pollute the dns with this method... Well, this may just be a conspiracy theory idea.
I hope dice2o can solve this problem, after all, global mode is a bit of trouble to use....

**Comment by Sean0717** [comment]
I have a same problem

**Comment by QwindF** [comment]
Is there a solution now

**Comment by Marshall1115** [comment]
> > > nice, I have also just found this way to solve this problem, just set clash VPN proxy mode to global mode.
> 
> Perhaps it is some kind of commercial competition? 1.1.1.1 The provider of this dns is CloudFlare, which has a good relationship with Microsoft. If bing's competitors pollute the dns with this method... Well, this may just be a conspiracy theory idea. I hope dice2o can solve this problem, after all, global mode is a bit of trouble to use....

yes, It’s very likely to be the case,It might be a problem that only occurs in China

**Comment by lettered-olive** [comment]
I put it on global mode, still didn't work...

**Comment by dice2o** [comment]
Fixed in v0.3.1.
It seems that Bing has added more rules to IP detection, and IPs from servers (like AWS, GCP) will be restricted. So even if the VPN is connected, Bing still shows this error.
I modify the headers of every request to bypass the server IP detection and temporarily solve the problem. Or you can connect to a node that has a local IP.

**Comment by Deep1994** [comment]
> modify the headers of every request to bypass the server IP

Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?

**Comment by dice2o** [comment]
> > modify the headers of every request to bypass the server IP
> 
> Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?

This feature is already in v0.3.1, download it from release page.
If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

**Comment by Deep1994** [comment]
> 

It works! Thank you very much!

**Comment by l6l6ng** [comment]
> > > modify the headers of every request to bypass the server IP
> > 
> > 
> > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> 
> This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

It works for me! tks

**Comment by 756180** [comment]
> > > modify the headers of every request to bypass the server IP
> > 
> > 
> > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> 
> This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

Hey mate, any thoughts on Shadowrocket? how do you use it on iOS? Appreciate for your efforts, btw thanks for app!

**Comment by Entropic0blivion** [comment]
I have this same problem :/

**Comment by dice2o** [comment]
> In mainland China, you can have a conversation with him as long as you get the page with vpn first. But recently, I found that I also need to open vpn when I have a conversation with him for the first time. I don’t like to open the global proxy, but I still hope that the software can set the proxy separately. [Machine translation]
> 
> 在中国大陆，之前只要先用vpn获取到页面就可以和他对话。但是最近，我发现我第一次和他对话的时候也需要打开vpn,我比较不喜欢打开全局代理，但我还是希望软件能单独设置代理。

Yes, because Bing also started IP detection for chat requests. Can you customize rules in VPN software? Just let bing.com and its subdomains be correctly proxied, and don't let your local DNS resolve these domains.

**Comment by dice2o** [comment]
> > > > modify the headers of every request to bypass the server IP
> > > 
> > > 
> > > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> > 
> > 
> > This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .
> 
> Hey mate, any thoughts on Shadowrocket? how do you use it on iOS? Appreciate for your efforts, btw thanks for app!

Check if your VPN app supports scripts that modify headers. If not, it is better to connect to a node with a local IP that is not marked as a server by Bing

**Comment by Ksorz** [comment]
> > > modify the headers of every request to bypass the server IP
> > 
> > 
> > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> 
> This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

Solution is works, but not for the sidebar chat in Edge. Is seems the modheader plugin does not change the header for sidebar window. Is it possible to make it work?

**Comment by 756180** [comment]
Appreciate it but quite confusing

Would you mind to drop me an invite in weixin, would be great to get a hand of help if you are open: andrew_tiosa

**Comment by nwaobasianthony** [comment]
> > > modify the headers of every request to bypass the server IP
> > 
> > 
> > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> 
> This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

worked for me, thanks

**Comment by behroozbc** [comment]
> > > > nice, I have also just found this way to solve this problem, just set clash VPN proxy mode to global mode.
> > 
> > 
> > Perhaps it is some kind of commercial competition? 1.1.1.1 The provider of this dns is CloudFlare, which has a good relationship with Microsoft. If bing's competitors pollute the dns with this method... Well, this may just be a conspiracy theory idea. I hope dice2o can solve this problem, after all, global mode is a bit of trouble to use....
> 
> yes, It’s very likely to be the case,It might be a problem that only occurs in China

It have happened in Iran too

**Comment by introspection3** [comment]
is  my dear microsoft becomming a red dragon?

**Comment by iHamid** [comment]
> > > modify the headers of every request to bypass the server IP
> > 
> > 
> > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> 
> This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

It works for me! I am from Iran and I have to use VPN. Thanks

**Comment by babakad** [comment]
> > > > modify the headers of every request to bypass the server IP
> > > 
> > > 
> > > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> > 
> > 
> > This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .
> 
> It works for me! I am from Iran and I have to use VPN. Thanks

mishe bgin chikar bayad bokonam ta dorost she chon in moshkelo man ham chand rooze daram...

**Comment by behroozbc** [comment]
یه افزونه هستش به اسم modheader نصب کردیم.
x-forwarded-for  1.1.1.1یه هدر هم تنظیم کردیم که
به نظرم تو فید بک هاش هی بنویسم که مشکل خوردیم و نمیتونیم اینا تا این رو درست کنه

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by dice2o** [comment]
> > > > > > modify the headers of every request to bypass the server IP
> > 
> > 
> > > > >
> > 
> > 
> > > > >
> > 
> > 
> > > > > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .
> > 
> > 
> > > 
> > 
> > 
> > > Hey mate, any thoughts on Shadowrocket? how do you use it on iOS? Appreciate for your efforts, btw thanks for app!
> > 
> > 
> > Check if your VPN app supports scripts that modify headers. If not, it is better to connect to a node with a local IP that is not marked as a server by Bing
> 
> Appreciate it but quite confusing
> 
> Would you mind to drop me an invite in weixin, would be great to get a hand of help if you are open: andrew_tiosa

Sorry, I didn't find a good way on iOS either. Connect to a node that is not blocked by Bing.

**Comment by cssius111** [comment]
This issue started the week before last...

**Comment by freestylefly** [comment]
> > > > modify the headers of every request to bypass the server IP
> > > 
> > > 
> > > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> > 
> > 
> > This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .
> 
> It works for me! I am from Iran and I have to use VPN. Thanks

thanks，I am OK

**Comment by GXOmar** [comment]
> > > modify the headers of every request to bypass the server IP
> > 
> > 
> > Hi, could you talk more about how you modify the headers of every request? With what tool and how to modify?
> 
> This feature is already in v0.3.1, download it from release page. If you use browser to visit bing, use a plugin, such as modheader, to change `x-forwarded-for` to `1.1.1.1` .

Thanks, it worked, I encountered this issue while using a VPN.

**Comment by Bprataap** [comment]
This error is primarily experienced by users in China. To resolve this, you can try the following potential worarounds:

1. Change DNS
2. Disable Extensions
3. Enable or Disable VPN
4. Use Edge Dev Version

Read about these in detail here ([Fix Bing Chat "Sorry, looks like your network settings are preventing access to this feature](https://mefmobile.org/bing-chat-sorry-looks-like-your-network-settings-are-preventing-access/))

**Comment by Dragon1573** [comment]
I also meet the same problem. All above 4 solutions can't fix it on my device.

## Environments

- User location: **Mainland China**
- DNS server: `192.168.2.1` (LAN) -> `1.1.1.1` & `8.8.8.8` (WAN)
- VPN / Proxy: **Enabled** (by _Clash for Windows_ )
- Proxy localtion: **Mumbai (India)**
- Proxy mode: **Global**
- Browser: **Microsoft Edge Dev (version 114.0.1788.0)** with all extensions disabled

## Configurations

- Microsoft Bing
  - Language: **Chinese (Simplified)**
  - Country or Region: **India**
  - Area: **Guangdong Province** (Mainland China)

**Comment by Dragon1573** [comment]
> So, does it mean this app is useless in china now ?

I'm able to interact with Bing with this project. Maybe BingGPT is using APIs reverse-engineer-ed from original Bing web page?

I can't directly use Bing via the official page even if I'm behind a proxy, but CAN via BingGPT or ChatHub with proxy enabled.
